3rdparty/libunwind/CMakeLists.txt

```diff
@@ -1,110 +1,153 @@
-else()
-    list (APPEND OPTS "--enable-debug=no" "--enable-debug-frame==no")
-endif()
-
-include (ExternalProject)
-
-ExternalProject_Add(oelibcxx_unwind
-
-    # Copy libunwind source tree to build directory.
-    DOWNLOAD_COMMAND ${CMAKE_COMMAND} -E copy_directory
-        ${CMAKE_CURRENT_LIST_DIR}/libunwind 
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind
-
-    # Remove -Wall as this is applied too late and will override -Wno-* flags in Clang.
-    # Instead we apply it ourselves using CFLAGS (see above).
-    COMMAND sed -e "s/-Wall//g"
-        ${CMAKE_CURRENT_LIST_DIR}/libunwind/configure.ac >
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/configure.ac
-
-    # Generate autoconf scripts and run configure.
-    CONFIGURE_COMMAND cd ${CMAKE_CURRENT_BINARY_DIR}/libunwind && 
-        ./autogen.sh "${OPTS}" "CFLAGS=${CFLAGS}"
-
-    # Copy libunwind-common.h to libunwind-common.inc.
-    COMMAND ${CMAKE_COMMAND} -E copy
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/include/libunwind-common.h
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/include/libunwind-common.inc
-
-    # Copy libunwind-common.h to the build directory (this file includes
-    # libunwind-common.inc).
-    COMMAND ${CMAKE_COMMAND} -E copy
-        ${CMAKE_CURRENT_LIST_DIR}/libunwind-common.h
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/include/libunwind-common.h
-
-    # Copy Gstep.c to Gstep.inc.
-    COMMAND ${CMAKE_COMMAND} -E copy
-        ${CMAKE_CURRENT_LIST_DIR}/libunwind/src/x86_64/Gstep.c
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/src/x86_64/Gstep.inc
-
-    # Copy Gstep.c to the the build directory.
-    COMMAND ${CMAKE_COMMAND} -E copy
-        ${CMAKE_CURRENT_LIST_DIR}/Gstep.c
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/src/x86_64/Gstep.c
-
-    # Copy Gtrace.c to Gtrace.inc while removing the line containing
-    # "static __thread".
-    COMMAND grep -v "^static __thread"
-        ${CMAKE_CURRENT_LIST_DIR}/libunwind/src/x86_64/Gtrace.c >
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/src/x86_64/Gtrace.inc
-
-    # Copy Gtrace.c to the the build directory.
-    COMMAND ${CMAKE_COMMAND} -E copy
-        ${CMAKE_CURRENT_LIST_DIR}/Gtrace.c
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/src/x86_64/Gtrace.c
-
-    # Copy setcontext.S over the one in the build directory.
-    COMMAND ${CMAKE_COMMAND} -E copy
-        ${CMAKE_CURRENT_LIST_DIR}/setcontext.S
-        ${CMAKE_CURRENT_BINARY_DIR}/libunwind/src/x86_64/setcontext.S
-
-    # Build libunwind.
-    BUILD_COMMAND $(MAKE) -C ${CMAKE_CURRENT_BINARY_DIR}/libunwind/src
-
-    # Empty install command.
-    INSTALL_COMMAND "")
+configure_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/libunwind/include/libunwind-x86_64.h
+  ${CMAKE_CURRENT_BINARY_DIR}/libunwind.h
+  COPYONLY)
+
+configure_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/libunwind/src/x86_64/Gstep.c
+  ${CMAKE_CURRENT_BINARY_DIR}/Gstep.inc
+  COPYONLY)
+
+add_custom_command(
+  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/Gtrace.c
+  COMMAND grep -v "^static __thread"
```

Comment:
For next iteration: We are doing this ugly workaround since we don't support thread local storage (__thread).
Once the PR for Thread Local Storage goes in (#1157 ), we can remove this.

Reply:
I will leave a TODO.